niviz/interfaces/views.py

```diff
@@ -48,3 +50,75 @@ def _post_run_hook(self, runtime):
 
     def _run_interface(self, runtime):
         return runtime
+
+
+class _ISegInputSpecRPT(nrc._SVGReportCapableInputSpec,
+                        BaseInterfaceInputSpec):
+    '''
+    Input specification for ISegRPT, implements:
+
+    anat_file: Input anatomical image
+    seg_files: Input segmentation image(s) - can be a list or a single file
+    mask_file: Input ROI mask
+
+    Bases _SVGReportCapableInputSpec which implements:
+
+    out_report: Filename trait
+    compress_report: ["auto", true, false]
+
+    '''
+    anat_file = File(exists=True,
+                     usedefault=False,
+                     resolve=True,
+                     desc='Anatomical image of SVG',
+                     mandatory=True)
+
+    seg_files = InputMultiPath(
+                    File(exists=True,
+                         usedefault=False,
+                         resolve=True),
+                    desc='Segmentation image of SVG',
+                    mandatory=True)
+
+    mask_file = File(exists=True,
+                     udedefault=False,
+                     resolve=True,
+                     desc='ROI Mask for mosaic',
+                     mandatory=True)
+
+
+class _ISegOutputSpecRPT(reporting.ReportCapableOutputSpec):
+    pass
+
+
+class ISegRPT(nrc.SegmentationRC):
+    '''
+    Class to generate registration images from pre-existing
+    NIFTI files. 
+
+    Effectively acts as an Identity node with report
+    generation as a side-effect.
+    '''
+
+    # Use our declared IO specs
+    input_spec = _ISegInputSpecRPT
+    output_spec = _ISegOutputSpecRPT
+
+    def _post_run_hook(self, runtime):
+        '''
+        Do nothing but propogate properties
+        to (first) parent class of ISegRPT
+        that is nrc.SegmentationRC
+        '''
+
+        # Set variables for `nrc.SegmentationRC`
+        self._anat_file = self.inputs.anat_file
+        self._seg_files = self.inputs.seg_files
+        self._mask_file = self.inputs.mask_file
+        self._masked = True
```

Comment:
First line deals with the error that pops up if self.inputs.mask_file isn't specified

```suggestion
        self._mask_file = self.inputs.mask_file or None
        self._masked = self.inputs.masked
```